xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-pageobjects/src/main/java/org/xwiki/administration/test/po/LocalizationAdministrationSectionPage.java

```diff
@@ -35,21 +35,24 @@
  */
 public class LocalizationAdministrationSectionPage extends AdministrationSectionPage
 {
-    @FindBy(xpath = "(//div[contains(@class, 'bootstrap-select')])[1]")
+    @FindBy(xpath = "//div[contains(@class, 'bootstrap-select')"
+        + " and ./select[@id='XWiki.XWikiPreferences_0_multilingual']]")
     private WebElement multiLingualSelect;
 
-    @FindBy(xpath = "(//div[contains(@class, 'bootstrap-select')])[3]")
+    @FindBy(xpath = "//div[contains(@class, 'bootstrap-select')"
+        + " and ./select[@id='XWiki.XWikiPreferences_0_default_language']]")
     private WebElement defaultLanguageSelect;
 
-    @FindBy(xpath = "(//div[contains(@class, 'bootstrap-select')])[2]")
+    @FindBy(xpath = "//div[contains(@class, 'bootstrap-select')"
+        + " and ../input[@id='XWiki.XWikiPreferences_0_languages']]")
     private WebElement supportedLanguagesSelect;
 
     public LocalizationAdministrationSectionPage()
     {
         super("Localization");
         waitUntilActionButtonIsLoaded();
         // Wait for asynchronous widgets to be loaded
-        getDriver().waitUntilElementIsVisible(By.cssSelector(".bootstrap-select"));
+        getDriver().waitUntilElementIsVisible(By.xpath("(//div[contains(@class, 'bootstrap-select')])[3]"));
```

Comment:
This "3" means there's several bootstrap select on the same xpath if I'm correct, a bit fragile no? Can't we use a better xpath?

Reply:
I agree, does it match `multiLingualSelect`, `defaultLanguageSelect`, or `supportedLanguagesSelect`?

Reply:
My understanding of XPaths is that this should match if there are three matching elements. The previous selector already matched after the first select was ready. I am not sure though that this is actually necessary as accessing the individual elements should already wait for them to become available.

Reply:
Well the elements can be available in the DOM but not visible, I think that's what happened when Guillaume introduced this wait in https://github.com/xwiki/xwiki-platform/commit/b1affcf708dd5511edb54fefd089021e9d191b4a

Reply:
Okay, but it doesn't make any sense imho to wait for the first bootstrap-select element to be visible. The first one is the simple one with the static three options, which is much faster than the other two. I know this for sure because I have seen screen recordings of failing tests where just this first select was initialized and the test started interacting with it and then started interacting with the other two when they became available.

Reply:
I have now added explicit waits for all three elements, tests passing in Firefox and Chrome.